### ISSUE TITLE
Set advanced search collapse to default as open with first five selectors listed for related lists

### ISSUE THREAD

**Issue by roamye** [project-lux/lux-frontend#201]
**Problem Description:**
When a user selects "show all X results" from an entity pages accordion or from the entity tab it will display the advanced search of results with the AS details collapsed. This can cause confusion for users as normally these are defaulted to open. (ex: selection of timelines, or when a user creates the search such as those in the AS help section)  

**Expected Behavior/Solution:**
Since these searches usually contain more than 5 multiple field/single field selections, it is expected for these AS to default as open for the first [5/6] rows with a "show more" option. 

**Requirements:**
- [ ] AS will default open with a cut off of the first [X] rows.
- [ ] When there are more than [X] rows, the "show more" option will appear below. 
- [ ] "show less" option will appear once all rows have been shown. 

**Needed for promotion:**
If an item on the list is not needed, it should be crossed off but not removed.

- [ ] Wireframe/Mockup - Mike
- [ ] Committee discussions - Sarah
- [ ] Feasibility/Team discussion - Sarah
- [ ] Backend requirements - TBD
- [ ] Frontend requirements- TBD
- [ ] Questions
- What should the cutoff number be? It was discussed to be either 5 or 6.
- Should it instead be the first multiple field? 
- Where is the best place to display the "show more" option? Under the "Add row" or would it come before the "add row"? 

**UAT/LUX Examples:**
- Selection of any 'search all X results' in accordion or in entity tab for record [Andy Warhol](https://lux.collections.yale.edu/view/person/34f4eec7-7a03-49c8-b1be-976c2f6ba6ba)
- [Long AS](https://lux.collections.yale.edu/view/results/works?q={%22OR%22:[{%22AND%22:[{%22createdBy%22:{%22id%22:%22https://lux.collections.yale.edu/data/person/6aab25df-680d-4ab7-8994-20418e10d4af%22}},{%22createdBy%22:{%22id%22:%22https://lux.collections.yale.edu/data/person/34f4eec7-7a03-49c8-b1be-976c2f6ba6ba%22}}]},{%22AND%22:[{%22createdBy%22:{%22id%22:%22https://lux.collections.yale.edu/data/person/6aab25df-680d-4ab7-8994-20418e10d4af%22}},{%22aboutAgent%22:{%22id%22:%22https://lux.collections.yale.edu/data/person/34f4eec7-7a03-49c8-b1be-976c2f6ba6ba%22}}]},{%22AND%22:[{%22aboutAgent%22:{%22id%22:%22https://lux.collections.yale.edu/data/person/6aab25df-680d-4ab7-8994-20418e10d4af%22}},{%22createdBy%22:{%22id%22:%22https://lux.collections.yale.edu/data/person/34f4eec7-7a03-49c8-b1be-976c2f6ba6ba%22}}]},{%22AND%22:[{%22aboutAgent%22:{%22id%22:%22https://lux.collections.yale.edu/data/person/6aab25df-680d-4ab7-8994-20418e10d4af%22}},{%22aboutAgent%22:{%22id%22:%22https://lux.collections.yale.edu/data/person/34f4eec7-7a03-49c8-b1be-976c2f6ba6ba%22}}]}]}&openSearch=false)
- [short AS](https://lux.collections.yale.edu/view/results/events?q=%7B%22used%22%3A%7B%22containing%22%3A%7B%22producedBy%22%3A%7B%22id%22%3A%22https%3A%2F%2Flux.collections.yale.edu%2Fdata%2Fperson%2F34f4eec7-7a03-49c8-b1be-976c2f6ba6ba%22%7D%7D%7D%7D&sort=eventStartDate:asc&openSearch=false&es=eventStartDate:asc)

**Dependencies/Blocks:**

- Blocked By: Issues that are blocking the completion of the current issue.
- Blocking: Issues being blocked by the completion of the current issue.

**Related Github Issues:**

- Issues that contain similar work but are not blocking or being blocked by the current issue.

**Related links:**
- Discussion on [AS not populating by related list items links](https://teams.microsoft.com/l/message/19:GmLeQu1cBR40mfrno-l8lngg7RFCcJs8fvFIw-MJjIM1@thread.tacv2/1713887622851?tenantId=dd8cbebb-2139-4df8-b411-4e3e87abeb5c&groupId=4620eac6-0955-4c69-93e8-b493b21f0df6&parentMessageId=1713887622851&teamName=MarkLogic&channelName=General&createdTime=1713887622851). 

**Wireframe/Mockup:**
![image](https://github.com/project-lux/lux-frontend/assets/163148482/b9c17e31-da0c-4a48-85f9-520e528b4f0d)

![image](https://github.com/project-lux/lux-frontend/assets/163148482/cdf8abf3-5957-4719-b1c4-00ade5da32b2)

**Comment by prowns** [comment]
Notes from 4/26 team meeting:

- Rescope ticket to NOT do show more
- Default open for short ones, and close on long ones
- Think about if this is impacted by new POW designs

**Comment by kamerynB** [comment]
Questions:
Is the frontend to determine what is considered a "long" query? If so, would it be by the query URL string length?

**Comment by prowns** [comment]
@kamerynB - I believe it was longer than 5 lines of Adv. Search.  Is that possible to determine before rendering?

**Comment by kamerynB** [comment]
@prowns I can do that but what does that mean exactly? 5 lines? There is a lot of JSON to be parsed and should it consider nested JSON? What are we calling a "line"? 

Can this be discussed at forming? We already have collapse buttons but then we would be adding a "Show More" or "Show Less" button? This seems redundant.

**Comment by prowns** [comment]
@kamerynB - how about if there are three or fewer HAL links, render as open by default?  Otherwise, show collapsed.  Good to discuss at ticket forming

![image](https://github.com/project-lux/lux-frontend/assets/140099762/b1530aa0-5f35-4bcb-a21e-4bc1c74a8278)

**Comment by prowns** [comment]
@kamerynB - should we discuss https://www.bugherd.com/projects/284041/tasks/2287 with this ticket as well?

**Comment by kamerynB** [comment]
@prowns That bugherd is a separate issue. I don't think there is anything to discuss with that other than we need to add another button saying something that describes the action of wrapping the current search in another group, if that makes sense.

**Comment by roamye** [comment]
@kamerynB @prowns - was this discussed at last weeks ticket forming? it wasn't on the previous one (week of may 21). Have we decided on what the decision will be for the length of a query to be minimized/expanded?

**Comment by kamerynB** [comment]
@roamye This was discussed at some point but I don't think we came to a consensus.

**Comment by roamye** [comment]
From 6/14 team meeting today: 
A decision has been made to ...
- show the first 6 rows of any AS query that has more than 6 rows.
- add a "show all" button for those with more than 6 rows. 
- If a query has six rows or less then it will NOT show the show all button. 
- the show all button will be on the left hand side below the sixth row. 
- A row counts as anything with a trash can icon next to it.

**Comment by roamye** [comment]
Mockup has been updated & moved this to priotization review. Let me know if any changes 2 the mockup are needed. 

@kamerynB @prowns

**Comment by roamye** [comment]
from uat 11/6 this has been moved to the next milestone to give @kamerynB to make adjustments to the layout.

**Comment by roamye** [comment]
Approved by UAT

**Comment by roamye** [comment]
Looks good in PROD. closing.
